internal.go

```diff
@@ -0,0 +1,65 @@
+package config
+
+type Internal struct {
+	Bitswap InternalBitswap
+}
+
+const (
+	DefaultBitswapEngineBlockstoreWorkerCount = 2000
+	DefaultBitswapTaskWorkerCount             = 500
+	DefaultBitswapEngineTaskWorkerCount       = 500
+	DefaultBitswapMaxOutstandingBytesPerPeer  = 1 << 21
+)
+
+// InternalBitswap contains knobs for tuning bitswap resource utilization.
+// The knobs (below) document how their value should related to each other.
+// Whether their values should be raised or lowered should be determined
+// based on the metrics active_tasks, pending_tasks, pending_block_tasks and active_block_tasks
+// reported by bitswap.
+//
+// The value of active_tasks is capped by EngineTaskWorkerCount.
+//
+// The value of pending_tasks is generally capped by the knobs below,
+// however its exact maximum value is hard to predict as it depends on task sizes
+// as well as number of requesting peers. However, as a rule of thumb,
+// during healthy operation this value should oscillate around a "typical" low value
+// (without hitting a plateau continuously).
+//
+// If pending_tasks grows and eventually reaches a plateau,
+// while at the same time active_tasks is at its maximum,
+// the node has reached its resource limits and requests are being dropped (or not serviced).
+// Raising resource limits (using the knobs below) could help, assuming the hardware can support the new limits.
+//
+// The value of active_block_tasks is capped by EngineBlockstoreWorkerCount.
+//
+// The value of pending_block_tasks is indirectly capped by active_tasks, but cannot be predicted
+// as it depends on the number of blocks involved in a peer task which can vary.
+//
+// If the value of pending_block_tasks is observed to grow,
+// while active_block_tasks is at its maximum, there is indication that the number of
+// available block tasks is creating a bottleneck (either due to high-latency block operations,
+// or due to high number of block operations per bitswap peer task).
+// In such cases, try increasing the EngineBlockstoreWorkerCount.
+type InternalBitswap struct {
+	// number of threads sending outgoing messages.
+	// used to thottle the number of concurrent send operations.
+	TaskWorkerCount int
```

Comment:
idk how I feel about using 0 = default. We can do it if we document it as in https://github.com/ipfs/go-ipfs/blob/master/docs/config.md#autonatthrottleinterval.

It seems like we may want another config type like @Stebalien did for https://github.com/ipfs/go-ipfs/blob/master/docs/config.md#strings and https://github.com/ipfs/go-ipfs-config/blob/master/types.go where we can define the defaults, but I can also see pushing this off if it's a pain.

Reply:
Leaving this to 0 implies use the default defined in bitswap.go (as per your other comment), which is 8.

Reply:
+1 to using a new special numeric type (that allows for 0 as a valid value) and explicitly defined default via `WithDefault` 
My understanding is that we want default to be indicated by  "null" (or empty).

Reply:
Take care of in #142